internal/blockchain/common/common.go

```diff
@@ -147,13 +148,21 @@ func (cb *callbacks) BatchPinOrNetworkAction(ctx context.Context, nsOrAction str
 	// For V1 of the FireFly contract, namespace is passed explicitly, but needs to be mapped to local name(s).
 	// For V2+, namespace is inferred from the subscription.
 	if subInfo.Version == 1 {
-		namespaces := subInfo.V1Namespace[nsOrAction]
+		networkNamespace := params.NsOrAction
+		namespaces := subInfo.V1Namespace[networkNamespace]
 		if len(namespaces) == 0 {
-			log.L(ctx).Errorf("No handler found for blockchain batch pin on remote namespace '%s'", nsOrAction)
+			log.L(ctx).Errorf("No handler found for blockchain batch pin on network namespace '%s'", networkNamespace)
 			return nil
 		}
 		return cb.batchPinComplete(ctx, namespaces, batch, signingKey)
 	}
+	batch.TransactionType = core.TransactionTypeBatchPin
+	if strings.HasPrefix(params.NsOrAction, blockchain.FireFlyActionPrefix) {
+		typeName := params.NsOrAction[len(blockchain.FireFlyActionPrefix):]
+		if typeName == "contract_invoke_pin" {
```

Comment:
The "action" field from the event is used for many different things at this point. Here, if the action is "firefly:contract_invoke_pin", we interpret that as the transaction type (otherwise we fall back to the default transaction type of "batch_pin").